agr_literature_service/lit_processing/data_ingest/dqm_ingest/utils/md5sum_utils.py

```diff
@@ -162,6 +161,133 @@ def save_s3_md5data(md5dict, mods):
+            mod_id = None
+        else:
+            logger.error("mod not in database yet:" + mod)
+            continue
+        mod_id_sql = "mod_id is null" if mod_id is None else f"mod_id = {mod_id}"
+        for primary_key in md5sums:
+            if primary_key.isnumeric():
+                reference_id = primary_key
+            else:
+                reference_results = db_session.execute(f"select reference_id from cross_reference r where r.curie = '{primary_key}' and is_obsolete='false' ")
+                refs = reference_results.fetchall()
+                if len(refs) == 0 :
+                    continue
+                reference_id = refs[0]["reference_id"]
+            md5sum_results = db_session.execute(f"select md5sum, reference_mod_md5sum_id from reference_mod_md5sum where {mod_id_sql} and reference_id = {reference_id} ")
+            md5sums = md5sum_results.fetchall()
+            if len(md5sums) == 0:
+                print("insert new md5sum: " + mod + " primary_key:" + str(primary_key) + ' ' + md5dict[mod][primary_key])
+                if mod_id is None:
+                    db_session.execute(f"insert into reference_mod_md5sum(reference_id, md5sum, date_updated) values ({reference_id}, '{md5dict[mod][primary_key]}', 'now()') ")
+                else:
+                    db_session.execute(f"insert into reference_mod_md5sum(reference_id, mod_id, md5sum, date_updated) values ({reference_id}, {mod_id}, '{md5dict[mod][primary_key]}', 'now()') ")
+                continue
+            md5sum = md5sums[0]["md5sum"]
+            if (md5sum == md5dict[mod][primary_key]):
+                continue
+            else:
+                print('need to update: ' + mod + '->' + str(primary_key) + ' old:' + md5sum + '->new:' + md5dict[mod][primary_key])
+                reference_mod_md5sum_id = md5sums[0]["reference_mod_md5sum_id"]
+                try:
+                    db_session.execute(f"update reference_mod_md5sum set md5sum='{md5dict[mod][primary_key]}' where  reference_mod_md5sum_id ='{reference_mod_md5sum_id}' ")
+                except Exception as e:
+                    logger.error('Error: ' + str(type(e)))
+    db_session.commit()
+    db_session.close()
+
+
+def load_database_md5data(mods):
+    """
+
+    :param mods:["FB", "WB", "ZFIN", "SGD", "MGI", "RGD", "XB", "PMID"]
+    :return sample:
+    md5sum_data = {"XB": {
+                    "PMID:9241": "TEST1-9177c6f32fb8a80ef5955543b9dafde6",
+                    "PMID:10735": "TEST2-5a24bf3a9e634fab93122b7c45a5d326"
+                    },
+                    "FB": {
+                     "FB:FBrf0251347": "TEST3-d70b2ce7c56deab14722fb4ac2e7d287",
+                     "FB:FBrf0251348": "9ca72344a115c9ce612cab87869ccd94"
+                    },
+                    "PMID": {
+                    "PMID:9241": "TEST5-6eac9538fafd9f73eff28dd0a28a2edf"
+                    }
+            }
+    """
+
+    md5dict = {}
+    mod_ids = {}
+    try:
```

Comment:
do we need this try except? If so you should move the initialization of db_session outside it otherwise if there's an exception the session may be None

Reply:
Also, you can rewrite line 259-264 with a one liner: `db_session = create_postgres_session(False)
    mod_ids = {mod_abbr_id["abbreviation"]: mod_abbr_id["mod_id"] for mod_abbr_id in db_session.execute("select abbreviation, mod_id from mod")}`

Reply:
I don't think we need try/except. They should work so I get rid of it. Good catch! Thanks @valearna